integrationTests/testInitializer.go

```diff
@@ -89,6 +89,116 @@ func CreateMessengerWithKadDht(ctx context.Context, initialAddr string) p2p.Mess
 	return libP2PMes
 }
 
+// CreateMessengerWithNoDiscovery creates a new libp2p messenger with no peer discovery
+func CreateMessengerWithNoDiscovery(ctx context.Context) p2p.Messenger {
+	prvKey, _ := ecdsa.GenerateKey(btcec.S256(), rand.Reader)
+	sk := (*libp2pCrypto.Secp256k1PrivateKey)(prvKey)
+
+	libP2PMes, err := libp2p.NewNetworkMessengerOnFreePort(
+		ctx,
+		sk,
+		nil,
+		loadBalancer.NewOutgoingChannelLoadBalancer(),
+		discovery.NewNullDiscoverer(),
+	)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	return libP2PMes
+}
+
+// CreateFixedNetworkOf7Peers assembles a network as following:
+//
+//                             0------------------- 1
+//                             |                    |
+//        2 ------------------ 3 ------------------ 4
+//        |                    |                    |
+//        5                    6                    7
+func CreateFixedNetworkOf7Peers() ([]p2p.Messenger, error) {
```

Comment:
there are 8 effective peers in this test.

Reply:
Right :D